consensus/state.go

```diff
@@ -979,18 +979,22 @@ func (cs *State) enterNewRound(height int64, round int32) {
 	logger := cs.Logger.With("height", height, "round", round)
 
 	if cs.Height != height || round < cs.Round || (cs.Round == round && cs.Step != cstypes.RoundStepNewHeight) {
-		logger.Debug(
-			"entering new round with invalid args",
-			"current", fmt.Sprintf("%v/%v/%v", cs.Height, cs.Round, cs.Step),
-		)
+		if logger.IsDebugging() {
+			logger.Debug(
+				"entering new round with invalid args",
+				"current", fmt.Sprintf("%v/%v/%v", cs.Height, cs.Round, cs.Step),
```

Comment:
might be better to just have height/round/step fields rather than the string

Reply:
laziness prevails